### ISSUE TITLE
Read error: list changed size during iteration

### ISSUE THREAD

**Issue by DarkWarden85** [kukulich/home-assistant-jablotron100#61]
Hi

I found the following error when checking my logs:

> This error originated from a custom integration.
> 
> Logger: custom_components.jablotron100
> Source: custom_components/jablotron100/jablotron.py:1027
> Integration: Jablotron 100 (documentation, issues)
> First occurred: 11:03:26 (1 occurrences)
> Last logged: 11:03:26
> 
> Read error: list changed size during iteration

This happens on the following version of home assistant:

- Home Assistant 2023.1.2
- Supervisor 2022.12.1
- Operating System 9.4
- Frontend 20230104.0 - latest

Any idea what this error could be? My alarm system seems to work fine though.

**Comment by kukulich** [comment]
Please find the complete log.

**Comment by DarkWarden85** [comment]
I tried downloading diagnostics data, but it doesn't work:

![image](https://user-images.githubusercontent.com/79642261/212064889-2f2bee8b-6961-4afa-aa9f-b1947d1a0e60.png)

This problem doesn't occur with any other integration I have.
I would really like to provide you with further data regarding this issue, but I have no idea how to do so

**Comment by DarkWarden85** [comment]
Finally I was able to download the diagnostics data. Turns out one of my devices had no type. After assigning the correct type, I coulkd finally download the diagnostics data:
[config_entry-jablotron100-4630613b72ccd6175a470733d7a98eaf.json.txt](https://github.com/kukulich/home-assistant-jablotron100/files/10402322/config_entry-jablotron100-4630613b72ccd6175a470733d7a98eaf.json.txt)

Maybe the undefined type was already the source of the problem?
